src/components/EditTextModal/EditTextModal.js

```diff
@@ -0,0 +1,156 @@
+import React, { useState, useEffect } from 'react';
+import classNames from 'classnames';
+import { useSelector, useDispatch } from 'react-redux';
+import { useTranslation } from 'react-i18next';
+import { Swipeable } from 'react-swipeable';
+import { FocusTrap } from '@pdftron/webviewer-react-toolkit';
+
+import defaultTool from 'constants/defaultTool';
+import core from 'core';
+import actions from 'actions';
+import selectors from 'selectors';
+import AutoResizeTextarea from 'components/NoteTextarea/AutoResizeTextarea';
+
+import './EditTextModal.scss';
+
+const EditTextModal = () => {
+  const [isDisabled, isOpen] = useSelector(state => [
+    selectors.isElementDisabled(state, 'editTextModal'),
+    selectors.isElementOpen(state, 'editTextModal'),
+  ]);
+
+  const [t] = useTranslation();
+  const dispatch = useDispatch();
+  const [newText, setNewText] = useState('');
+  // Note: core.getSelectedTextQuads() to store selected text coordinates in the following format
+  // ie. {1:[{x1: 81, x2: 553, x3: 553, x4: 81, y1: 608, y2: 608, y3: 592, y4: 592, }, {...}]}
+  // when storing in textCoordinates, we take off the wrapper object and only store array
+  const [textCoordinates, setTextCoordinates] = useState([]);
+
+  const closeModal = () => {
+    dispatch(actions.closeElement('editTextModal'));
+    setNewText('');
+    core.setToolMode(defaultTool);
+  };
+
+  useEffect(
+    () => {
+      if (isOpen) {
+        //  prepopulate selected text
+        setNewText(core.getSelectedText());
+        // convert and store selected text coordinates
+        storeConvertedPDFCoordinates();
+      }
+    },
+    [isOpen]
```

Comment:
It's good practice to have possible changing functions as dependency as well. So in this case `[isOpen, setNewText]`.

`storeConvertedPDFCoordinates` is better to define inside `useEffect` as it is only used there. Otherwise you need to set it as dependency and then you also need to make sure it's not change on every re-render. If you move it inside useEffect you don't have to worry about that.

Reply:
You don't have to add `setNewText` since React recognizes that it's a useState setter (if you use the react hooks eslint plugin, it will ignore `setNewText`

Reply:
Sure, that is the case. I just like to be more explicit and not leave things for the reader to guess which ones are implicitly handled.

Anyways `storeConvertedPDFCoordinates` should be still moved inside the effect

Reply:
Thanks for the suggestion, I have moved `storeConvertedPDFCoordinates` to the `useEffect` hook